src/nvim/normal.c

```diff
@@ -3867,6 +3867,23 @@ static void nv_mousescroll(cmdarg_T *cap)
       cap->count0 = 3;
       nv_scroll_line(cap);
     }
+  } else {
+    /* Horizontal scroll - only allowed when 'wrap' is disabled */
+    if (!curwin->w_p_wrap) {
+      int val;
+      int step = 6;
+
+      if (mod_mask & (MOD_MASK_SHIFT | MOD_MASK_CTRL)) {
+          step = curwin->w_width;
+      }
+
+      val = curwin->w_leftcol + (cap->arg == MSCR_RIGHT ? -step : +step);
+      if (val < 0) {
+          val = 0;
+      }
+
+      gui_do_horiz_scroll(val, true);
+    }
```

Comment:
This whole block is identical with the block added in edit.c. Can you move it to a static function in mouse.c, something like `mouse_scroll_horiz`. Then `gui_do_horiz_scroll` can be made static, too.